### ISSUE TITLE
Bug: new "compile docs" are very buggy

### ISSUE THREAD

**Issue by dromer** [plugdata-team/plugdata#1357]
Not sure if anyone has actually tested this, but it's really bad UX for me:

- Window has blocky corners
- Documentation starts at the bottom
- Most/many of the links don't open
- Window is floating and somehow sticks to all desktops (Linux)
- Window cannot be moved except for grabbing the top: blocks window manager hints?
- Window has resize handle, but can't resize (similar to https://github.com/plugdata-team/plugdata/issues/1302 ?)

And then there has been some editing done to the text which results it odd phrasing, duplication and key information removed that was previously there.

I'm not sure if this popup-window format is particularly useful compared to a proper full documentation page.

**Comment by timothyschoen** [comment]
Yeah, it's a WIP.

>> Window has blocky corners
>> Window cannot be moved except for grabbing the top: blocks window manager hints?
Those are problems with using non-compositing WMs, it does work with GNOME. I'll do a pass on those issues again soon, I guess.

>> Most/many of the links don't open

It appears the link hit boxes are shifted downward a bit on Linux. Should be fixable.

>> Window cannot be moved except for grabbing the top: blocks window manager hints

Might fix that later (but not the highest priority, since you can still just move the window)

>> And then there has been some editing done to the text which results it odd phrasing

Sorry, I'll see what information I accidentally omitted while rephrasing.

>> I'm not sure if this popup-window format is particularly useful compared to a proper full documentation page.

Pretty sure it is, I've introduced it for the Heavy component as a test run, eventually the whole website documentation will also be available like this. We can include examples that you can open directly from the documentation dialog, and it's easier to view plugdata and the docs side-by-side.

**Comment by dromer** [comment]
For the compile docs I think it'll be better to split everything up per compile type then to dump everything on a single page. Right now the balance between information and images is quite skewed as well.

The images are not as useful inside the program as they are on the website.

Btw the `warning` line is also broken inside plugdata

**Comment by timothyschoen** [comment]
Oh god I finally figured out the strange offset on links...

JUCE uses a different text layouting system internally on macOS. The macOS one sees spaces as characters, the Windows/Linux ones don't. So if I store the link position in the string by index, I need to remove all whitespace on Linux and Windows, but not on macOS.

**Comment by timothyschoen** [comment]
I think all this is fixed now? 

I'll still look at the content of my documentation a bit, but the original problems (including non-compositing issues) should be fixed

**Comment by dromer** [comment]
Yeah it's more stable now, and nice that it now stays open as you start filling in stuff in the compile dialog.

What I'm missing is all the information that was added and edited previously. Now there isn't really much useful info.
Also I don't think the screenshots are particularly useful when running inside the program (and the screenshots resizing with the dialog makes them pretty huge and ugly)
